### ISSUE TITLE
Unify paths

### ISSUE THREAD

**Issue by leandro-lucarella-sociomantic** [sociomantic-tsunami/makd#121]
In Makd there is, for historical reasons, `$T` which stores the absolute path to the root of the project, and then `$C` which does too for the current directory, in case `make` is invoked from a proect's subdirectory via `make -C test` for example.

This was never really used in MakD and the inconsistencies coming from users (and even `Makd.mak`) using different combinations of `$T`/`$C` and not using any prefix, makes that sometimes the same file is used via a absolute or a relative path, which for some analysis tools (like code coverage) can potentially end up being considered as different files.

I think we can simplify Makd by just completely ditching `$T` and `$C` and just assume `make` will be always use the root of the project as the working directory (i.e. requiring users if they want to use `make` from a subdirectory, they have to use `make -C <top-level>`.

**Comment by joseph-wakeling-sociomantic** [comment]
Two thoughts on this:

* where possible, we don't want the absolute path to be present in any generated files (e.g. think of libraries where debug messages describe private details of developers' personal workspace layout)

* isn't it possible to do some kind of path-compare and eliminate the prefix that `$T` has relative to what `$C` has?  (i.e. given any path, chop off `$T` from it if possible)

**Comment by leandro-lucarella-sociomantic** [comment]
> where possible, we don't want the absolute path to be present in any generated files (e.g. think of libraries where debug messages describe private details of developers' personal workspace layout)

Yeah, that's exactly what I'm proposing, to get rid of the absolute paths.

> isn't it possible to do some kind of path-compare and eliminate the prefix that $T has relative to what $C has? (i.e. given any path, chop off $T from it if possible)

Yes, there is already something doing exactly that, it's used by the "nice output" function, they print relative paths for everything. But if we ditch the absolute paths completely, we don't need to care about doing those transformations anymore anyway... :)

**Comment by leandro-lucarella-sociomantic** [comment]
https://github.com/sociomantic-tsunami/makd/blob/4209a062d9bf0fd08aaae2541e9115ec0f6f17bc/Makd.mak#L320-L326
